yt_dlp/extractor/huya.py

```diff
@@ -131,3 +136,120 @@ def encrypt(self, params, stream_info, stream_name):
+            'uploader': '卡尔',
+            'uploader_id': '367138632',
+            'upload_date': '20210811',
+            'view_count': int,
+            'comment_count': int,
+            'like_count': int,
+        }, 'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url: str) -> dict:
+        video_id = self._match_id(url)
+        api_url = f'https://liveapi.huya.com/moment/getMomentContent?videoId={video_id}'
+
+        try:
+            response = self._download_json(api_url, video_id)
+        except Exception as e:
+            raise ExtractorError(f'Failed to download JSON data: {e}', expected=True)
+
+        video_data = traverse_obj(response, ('data', 'moment', 'videoInfo'))
+
+        if not isinstance(video_data, dict):
+            raise ExtractorError('No video data found')
+
+        formats = []
+        for definition in video_data.get('definitions', []):
+            format_info = {
+                'format_id': definition.get('defName'),
+                'url': definition.get('url'),
+                'ext': determine_ext(definition.get('url', '')),
+                'width': int_or_none(definition.get('width')),
+                'height': int_or_none(definition.get('height')),
+                'filesize': int_or_none(definition.get('size')),
+                'protocol': 'http',  # Assuming HTTP protocol
+                'format_note': f'{definition.get("defName", "")} - {definition.get("width", "")}x{definition.get("height", "")}',
+            }
+            formats.append(format_info)
+
+        return {
+            'id': video_id,
+            'title': video_data.get('videoTitle', 'Untitled'),
+            'thumbnail': video_data.get('videoCover'),
+            'formats': formats,
+            'duration': self._parse_duration(video_data.get('videoDuration', '0')),
+            'uploader': video_data.get('nickName'),
+            'uploader_id': str_or_none(video_data.get('uid')),
+            'upload_date': self._parse_date(video_data.get('videoUploadTime', '')),
+            'view_count': video_data.get('videoPlayNum'),
+            'comment_count': video_data.get('videoCommentNum'),
+            'like_count': video_data.get('favorCount'),
+        }
+
+    def _parse_duration(self, duration_str: str) -> int:
+        """Parse duration string (e.g., '00:14' or '01:23:45') into seconds."""
+        try:
+            duration = parse_duration(duration_str)
+            return int(duration) if duration is not None else 0
+        except ValueError:
+            return 0
+
+    def _parse_date(self, date_str: str) -> str:
+        """Convert date string to YYYYMMDD format using yt-dlp's unified_strdate method."""
+        try:
+            # Use yt-dlp's unified_strdate method to parse the date string
+            parsed_date = unified_strdate(date_str)
+            return parsed_date if parsed_date is not None else ''
+        except ValueError:
+            return ''
```

Comment:
Let's simplify this by using `traverse_obj` from `..utils.traversal`
```suggestion
        return {
            'id': video_id,
            'formats': formats,
            **traverse_obj(video_data, {
                'title': ('videoTitle', {str}),
                'thumbnail': ('videoCover', {url_or_none}),
                'duration': ('videoDuration', {parse_duration}),
                'uploader': ('nickName', {str}),
                'uploader_id': ('uid', {str_or_none}),
                'upload_date': ('videoUploadTime', {unified_strdate}),
                'view_count': ('videoPlayNum', {int}),
                'comment_count': ('videoCommentNum', {int}),
                'like_count': ('favorCount', {int}),
            }),
        }
```